carla/recourse_methods/catalog/roar/library/roar.py

```diff
@@ -0,0 +1,196 @@
+import datetime
+from typing import List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.optim as optim
+from scipy.optimize import linprog
+from torch import nn
+from torch.autograd import Variable, grad
+
+from carla import log
+from carla.recourse_methods.processing import reconstruct_encoding_constraints
+
+
+def calc_max_perturbation(
```

Comment:
This can be `_calc_max_perturbation` right? That would make it clear this function is only use locally in this file.